pkg/controller/machine/machine_controller_test.go

```diff
@@ -17,87 +17,217 @@ limitations under the License.
-	// channel when it is finished.
-	mgr, err := manager.New(cfg, manager.Options{
-		Metrics: server.Options{
-			BindAddress: "0",
-		},
-	})
-	if err != nil {
-		t.Fatalf("error creating new manager: %v", err)
-	}
-	c = mgr.GetClient()
+		By("Creating the Machine")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
 
-	a := newTestActuator()
-	recFn := newReconciler(mgr, a)
-	if err := add(mgr, recFn, "dummy"); err != nil {
-		t.Fatalf("error adding controller to manager: %v", err)
-	}
+		By("Setting the AuthoritativeAPI to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMachineAPI
+		})).Should(Succeed())
+
+		Eventually(func() (machinev1.MachineStatus, error) {
+			machine := &machinev1.Machine{}
+			err := k8sClient.Get(ctx, objectKey(instance), machine)
+			if err != nil {
+				return machinev1.MachineStatus{}, err
+			}
+			return machine.Status, nil
+		}, timeout).ShouldNot(Equal(machinev1.MachineStatus{}))
 
-	stop, errChan := StartTestManager(mgr, t)
-	defer func() {
-		stop()
-		if err := <-errChan; err != nil {
-			t.Fatalf("error starting test manager: %v", err)
+		// // Expect platform status is not empty. This (check) means we've called the
+		// // actuator
+		// Eventually(func() (runtime.RawExtension, error) {
+		// 	machine := &machinev1.Machine{}
+		// 	err := k8sClient.Get(ctx, objectKey(instance), machine)
+		// 	if err != nil {
+		// 		return runtime.RawExtension{}, err
+		// 	}
+		// 	return *machine.Status.ProviderStatus, nil
+		// }, timeout).ShouldNot(Equal(runtime.RawExtension{}))
+
+		// TODO: Work out how to make test actuator set the provider status
+
+	})
+
+	It("Should set the Paused condition appropriately", func() {
+		instance := &machinev1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "foo",
+				Namespace:    "default",
+				Labels: map[string]string{
+					machinev1.MachineClusterIDLabel: "foo",
+				},
+			},
+			Spec: machinev1.MachineSpec{
+				ProviderSpec: machinev1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte("{}"),
+					},
+				},
+			},
 		}
-	}()
 
-	// Create the Machine object and expect Reconcile and the actuator to be called
-	if err := c.Create(context.TODO(), instance); err != nil {
-		t.Fatalf("error creating instance: %v", err)
+		By("Creating the Machine")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
+
+		By("Setting the AuthoritativeAPI to ClusterAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityClusterAPI
+		})).Should(Succeed())
+
+		By("Verifying that the AuthoritativeAPI is set to Cluster API")
+		Eventually(func() (machinev1.MachineAuthority, error) {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return "", err
+			}
+			return instance.Status.AuthoritativeAPI, nil
+		}, timeout).Should(Equal(machinev1.MachineAuthorityClusterAPI))
```

Comment:
Could use `k.Get` and `HaveField` here as well, you can do `HaveField("Status.MachineAuthority", ...)` and change the fields as well